### DISCUSSION TITLE
Deeper Diagrams.net Integration

### DISCUSSION THREAD

**Discussion by routerino** [outline/outline#2827] [Ideas]
Hi,

I saw that Diagrams.net integration existed in outline, and that's cool! However, I noticed that the integration isn't all it could be: it simply takes an external diagrams.net URL to display the content (as an iframe I assume).

Dragrams.net's [integration features](https://www.diagrams.net/integrations) (EDIT: [technical documentation here](https://www.diagrams.net/doc/faq/embed-mode)) allows for PNGs with embedded diagrams.net data to be used as a data source. This means that Outline, rather than outsourcing the document generation, could instead use self hosted images. This means no URL needs to be specified and editing of the diagram can be done without leaving the application.

You can see this integration in action with bookstack:
![wfc2v3BDUv](https://user-images.githubusercontent.com/45954722/144963657-72e32a31-73c3-408c-80eb-35f433c9f2ba.gif)

Would this be considered as an option for a feature request?
(PS: if yes, also add an environment variable for the Diagrams.net URL. Some of us self-host diagrams.net as well).

**Comment by johnoscott** [comment]
I think this is a good general usability approach for editing embed contents within Outline.

**Reply by routerino** [reply]
sounds good, I'll turn it into a feature request

**Comment by jeffshead** [comment]
Still not implemented? I REALLY like Outline but the lack of integration of self-hosted draw.io is a deal breaker for me.
